grakn-core/src/main/java/ai/grakn/concept/ConceptId.java

```diff
@@ -59,15 +60,22 @@ public Object getRawValue(){
     public boolean equals(Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
-
-        ConceptId conceptId1 = (ConceptId) o;
-
-        return getValue().equals(conceptId1.getValue());
+        ConceptId cast = (ConceptId) o;
+        if (conceptId instanceof String) {
```

Comment:
The concept ID is passed around as a string to other layers which leads to the concept ID sometimes being a string and sometimes being whatever the native ID type is (long for tinkerpop for example) so we end up comparing strings to longs with .equals and it fails. And instanceof check + long comparison is quicker than creating a string object, copying a long into it and then comparing the longs. A UUID is made up of two longs, so ideally it should also be compared as such rather than a string. We got 50% performance boost out of that optimization!